src/OrbitGl/CaptureWindowTest.cpp

```diff
@@ -79,4 +81,193 @@ TEST_F(UnitTestCaptureWindow, SlidersRespondToMouseLeave) {
+    // viewport height should be changed and the test adjusted accordingly
+    assert(time_graph_->GetTimelineUi()->GetHeight() < 95);
+    assert(time_graph_->GetTimelineUi()->GetPos()[0] == 0);
+  }
+
+ protected:
+  void ExpectInitialState() {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(vertical_slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(vertical_slider_->GetPosRatio(), 0.0);
+
+    EXPECT_EQ(slider_->GetLengthRatio(), 1.0);
+    EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+    EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+    EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+  }
+
+  enum struct PosWithinCapture { kLeft, kRight, kInBetween };
+  void ExpectIsHorizontallyZoomedIn(PosWithinCapture pos) {
+    UpdateChildrenPosAndSize();
+
+    EXPECT_LT(slider_->GetLengthRatio(), 1.0);
+    switch (pos) {
+      case PosWithinCapture::kLeft:
+        EXPECT_EQ(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_EQ(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kRight:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_EQ(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+      case PosWithinCapture::kInBetween:
+        EXPECT_GT(slider_->GetPosRatio(), 0.0);
+        EXPECT_LT(time_graph_->GetMaxTimeUs() * 1000, time_graph_->GetCaptureMax());
+        EXPECT_GT(time_graph_->GetMinTimeUs() * 1000, 0);
+        break;
+    }
+  }
+
+ private:
+  CaptureClientAppInterfaceFake capture_client_app_fake_;
+  std::unique_ptr<orbit_client_data::CaptureData> capture_data_;
+
+  void CreateSchedulerTimers() {
+    using orbit_client_protos::TimerInfo;
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(TrackTestData::kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+    time_graph_->ProcessTimer(timer, nullptr);
+  }
+};
+
+TEST_F(NavigationTestCaptureWindow, ZoomTimeWorksAsExpected) {
+  ExpectInitialState();
+
+  const Vec2i kTimeGraphSize = viewport_.WorldToScreen(time_graph_->GetSize());
+
+  // Zoom time: Zoom in and out horizontally, assure we've zoomed in, and zoomed out
+  // again
+  // Test 1: Middle of the capture window
+  int x = kTimeGraphSize[0] / 2;
+  int y = kTimeGraphSize[1] - 5;
+  MouseWheelMoved(x, y, 1, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  MouseWheelMoved(x, y, -1, false);
+  ExpectInitialState();
+
+  // Keyboard
+  MouseMoved(x, y, false, false, false);
+  KeyPressed('W', false, false, false);
+  ExpectIsHorizontallyZoomedIn(PosWithinCapture::kInBetween);
+  KeyPressed('S', false, false, false);
+  ExpectInitialState();
+
+  // Test 2: Left of the capture window
```

Comment:
Should this be a separate test case?

Reply:
I was wondering the same. Would make error reports a little neater.